### PR TITLE
[#2852] feat(core): Add relational backend for Role Entity

### PR DESCRIPTION
**Pull request by lw-yang**
### What changes were proposed in this pull request?

Add relational backend for Role Entity

[note]: wait for #3031  merged first

### Why are the changes needed?

Fix: #2852 

### Does this PR introduce _any_ user-facing change?

N/A

### How was this patch tested?

ut

### CODE REVIEW

File: core/src/main/java/com/datastrato/gravitino/storage/relational/mapper/UserMetaMapper.java [hunk 1]
```diff
@@ -20,11 +21,12 @@
  * href="https://mybatis.org/mybatis-3/getting-started.html"></a>
  */
 public interface UserMetaMapper {
-  String TABLE_NAME = "user_meta";
+  String USER_TABLE_NAME = "user_meta";
+  String USER_RELATION_TABLE_NAME = "user_role_rel";
```
[the remark this answers is not in the document]
Comment (lw-yang) [reply core/src/main/java/com/datastrato/gravitino/storage/relational/mapper/UserMetaMapper.java]: `listUsersByRoleId` in `UserMetaMapper`，need to join `user_role_rel` table

`listRolesByUserId` in `RoleMetaMapper`, query roles by userId, also need to join `user_role_rel` table

**Comment by lw-yang** [comment]
@yuqi1129 @yuqi1129 Could you help to review it again

**Comment by qqqttt123** [comment]
I have added a new field `securable_object_type` and change the name `securable_object_identifier` to `securable_object_full_name`.

**Comment by lw-yang** [comment]
> I have added a new field `securable_object_type` and change the name `securable_object_identifier` to `securable_object_full_name`.

i will add it

File: core/src/main/java/com/datastrato/gravitino/storage/relational/service/RoleMetaService.java [hunk 2]
```diff
@@ -56,7 +74,7 @@ public void insertRole(RoleEntity roleEntity, boolean overwritten) {
           roleEntity.namespace() != null
               && !roleEntity.namespace().isEmpty()
               && roleEntity.namespace().levels().length == 3,
-          "The identifier should not be null and should have three level.");
+          "The namespace of roleEntity should not be null and should have three level.");
```
Comment (qqqttt123) [inline core/src/main/java/com/datastrato/gravitino/storage/relational/service/RoleMetaService.java:77]: The namespace of RoleEntity must have three levels.

File: core/src/main/java/com/datastrato/gravitino/storage/relational/service/RoleMetaService.java [hunk 3]
```diff
@@ -79,4 +97,40 @@ public void insertRole(RoleEntity roleEntity, boolean overwritten) {
       throw re;
     }
   }
+
+  public RoleEntity getRoleByIdentifier(NameIdentifier identifier) {
+    Preconditions.checkArgument(
+        identifier != null
+            && !identifier.namespace().isEmpty()
+            && identifier.namespace().levels().length == 3,
+        "The identifier should not be null and should have three level.");
```
Comment (yuqi1129) [inline core/src/main/java/com/datastrato/gravitino/storage/relational/service/RoleMetaService.java:106]: ditto

File: core/src/main/java/com/datastrato/gravitino/storage/relational/service/RoleMetaService.java [hunk 4]
```diff
@@ -79,4 +97,40 @@ public void insertRole(RoleEntity roleEntity, boolean overwritten) {
...
+    RolePO rolePO = getRolePOByMetalakeIdAndName(metalakeId, identifier.name());
+
+    return POConverters.fromRolePO(rolePO, identifier.namespace());
+  }
+
+  public boolean deleteRole(NameIdentifier identifier) {
+    Preconditions.checkArgument(
+        identifier != null
+            && !identifier.namespace().isEmpty()
+            && identifier.namespace().levels().length == 3,
```
Comment (yuqi1129) [inline core/src/main/java/com/datastrato/gravitino/storage/relational/service/RoleMetaService.java:118]: +1

File: core/src/main/java/com/datastrato/gravitino/storage/relational/service/RoleMetaService.java (hunk 2 above)
Comment (lw-yang) [reply core/src/main/java/com/datastrato/gravitino/storage/relational/service/RoleMetaService.java]: user and group are also checked in this way.

 I would like to create another pr to solve this together. add `checkUser`,`checkGroup`,`checkRole` in `NameIdentifier`

Comment (lw-yang) [reply core/src/main/java/com/datastrato/gravitino/storage/relational/service/RoleMetaService.java]: WDYT

Comment (qqqttt123) [reply core/src/main/java/com/datastrato/gravitino/storage/relational/service/RoleMetaService.java]: OK.

**Review by yuqi1129 — APPROVED** [review]
LGTM

**Pull request merged**